packages/valory/contracts/gnosis_safe/contract.py

```diff
@@ -449,15 +451,51 @@ def get_raw_safe_transaction(  # pylint: disable=too-many-arguments,too-many-loc
             and max_priority_fee_per_gas is None
         ):
             tx_parameters.update(ledger_api.try_get_gas_pricing(old_price=old_price))
-        # note, the next line makes an eth_estimateGas call!
         transaction_dict = w3_tx.buildTransaction(tx_parameters)
-        transaction_dict["gas"] = Wei(
-            max(transaction_dict["gas"] + 75000, configured_gas)
+        estimated_gas = cls._estimate_gas(
+            ledger_api,
+            transaction_dict,
         )
+        transaction_dict["gas"] = Wei(max(estimated_gas + 75000, configured_gas))
         transaction_dict["nonce"] = nonce  # pragma: nocover
 
         return transaction_dict
 
+    @staticmethod
+    def _estimate_gas(ledger_api: EthereumApi, tx: Dict) -> int:
+        """Estimate the gas for the given tx."""
+        estimated_gas: Optional[int] = None
+        transaction_dict = deepcopy(tx)
+        # we delete the set gas to not interfere with the estimation
+        del transaction_dict["gas"]
+
+        try:
+            estimated_gas = ledger_api.api.eth.estimate_gas(transaction_dict)
+        except ContractLogicError as e:
+            _logger.warning(
+                "Failed to estimate gas with the default block identifier, "
+                f"{type(e).__name__}: {e.__str__()}"
+            )
+
+        if estimated_gas is None:
+            # estimation using default args may fail if the pending state (block)
+            # is used by the node, which might happen when repricing
+            try:
+                estimated_gas = ledger_api.api.eth.estimate_gas(
```

Comment:
this might throw an error, as some nodes don't accept the block identifier as arg.